tide/wall/PixelStreamUpdater.cpp

```diff
@@ -45,39 +45,35 @@
 #include "log.h"
 #include "network/WallToWallChannel.h"
 
-#include <deflect/Frame.h>
-#include <deflect/SegmentDecoder.h>
+#include <deflect/server/Frame.h>
+#include <deflect/server/TileDecoder.h>
 
 #include <QImage>
 #include <QThreadStorage>
 
 namespace
 {
-void _splitByView(const deflect::Segments& segments,
-                  deflect::Segments& leftOrMono, deflect::Segments& right)
+void _splitByView(const deflect::server::Tiles& tiles,
+                  deflect::server::Tiles& leftOrMono,
+                  deflect::server::Tiles& right)
 {
-    std::partition_copy(segments.begin(), segments.end(),
-                        std::back_inserter(right),
-                        std::back_inserter(leftOrMono),
-                        [](const deflect::Segment& segment) {
-                            return segment.view == deflect::View::right_eye;
+    std::partition_copy(tiles.begin(), tiles.end(), std::back_inserter(right),
+                        std::back_inserter(leftOrMono), [](const auto& tile) {
+                            return tile.view == deflect::View::right_eye;
                         });
     assert(right.empty() || right.size() == leftOrMono.size());
 }
 
-void _sortByPosition(deflect::Segments& segments)
+void _sortByPosition(deflect::server::Tiles& tiles)
 {
-    std::sort(segments.begin(), segments.end(),
-              [](const deflect::Segment& s1, const deflect::Segment& s2) {
-                  return (s1.parameters.y == s2.parameters.y
-                              ? s1.parameters.x < s2.parameters.x
-                              : s1.parameters.y < s2.parameters.y);
-              });
+    std::sort(tiles.begin(), tiles.end(), [](const auto& t1, const auto& t2) {
+        return (t1.y == t2.y ? t1.x < t2.x : t1.y < t2.y);
+    });
 }
 }
 
 PixelStreamUpdater::PixelStreamUpdater()
-    : _headerDecoder{new deflect::SegmentDecoder}
+    : _headerDecoder{new deflect::server::TileDecoder}
```

Comment:
make_shared/unique. There are actually more 'new' here in this PR, maybe you want to change them too.

Reply:
This is an interesting question of coding style. Naked 'new' are definitely bad, and I also tend to prefer re-assigning a pointer with = make_unique() instead of .reset(). But I don't know if we should go so far as banning all 'new'..? I've almost always seen it done that way to initialize member smart_ptrs. There's likely no performance implications when using make_unique in this situation. There might be a difference (however small) in doing make_shared because I think it allocates both the pointer's memory and object's memory in one go, so I don't know if it can be emplaced in the class or if it's moved..?